shared/workspaceService.ts

```diff
@@ -190,19 +202,22 @@ export function buildWorkspaceCriteria(
         dataSources: buildWorkspaceDataSources(
           criterion,
           idMapper(oldId),
-          idMapper
+          idMapper,
+          dataSourceOrder
         )
       };
     }
   );
+  return order(criteriaOrder, newCriteria);
```

Comment:
using 'order' as both verb and noun in the same file is a bit confusing. How about `applyOrdering` for the function?

Or preserve 'ing' above and make the distinction clearer that way. Or both :)